### PR TITLE
Automation improvements

### PR DESCRIPTION
**Pull request by kmcquade**
As I discussed with @0xdabbad00 offline, I will be making some PRs to CloudTracker here soon. In anticipation of that, I thought I'd add some small improvements to make review easier. This first PR adds the following:

* Modifies `setup.cfg` so it points to the correct tests and excludes certain tests from the coverage scan
* Adds `requirements.txt` and `requirements-dev.txt`
* Adds unit tests to GitHub actions
* Leverages PyInvoke (see the `tasks.py` file) for stashing common automation tasks. This makes it easier to run automation and testing commands within GitHub actions or otherwise without worrying about the specific command flags etc.
* Makes the `bin/cloudtracker` file executable for easier usage in PyInvoke.
* Adds some lines to `.gitignore`

I plan on sending a separate PR after I run `autopep8` (based on the `setup.py` file, it looks like that's what you prefer) and fix the various `pylint` errors. This way, you will have a clean point of comparison for where I make any modifications.

### CODE REVIEW

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -27,5 +27,5 @@ jobs:
         TWINE_USERNAME: ${{ secrets.PYPI_USERNAME }}
         TWINE_PASSWORD: ${{ secrets.PYPI_PASSWORD }}
       run: |
-        python setup.py sdist bdist_wheel
+        python -m setup.py sdist bdist_wheel
```
Comment (0xdabbad00) [inline .github/workflows/publish.yml:30]: Doesn't this try to load a module named `setup.py`?

File: requirements.txt [hunk 2]
```diff
@@ -0,0 +1,13 @@
+ansicolors==1.1.8
+boto3==1.5.32
+botocore==1.8.50
+docutils==0.16
+jmespath==0.9.3
+python-dateutil==2.8.1
+PyYAML==4.2b4
+s3transfer==0.1.13
+six==1.15.0
+elasticsearch[es1]==1.9.0
```
Comment (0xdabbad00) [inline requirements.txt:10]: I would rather not have the elasticsearch modules in the requirements.txt as that functionality is no longer maintained or recommended.

File: requirements.txt [hunk 3]
```diff
@@ -0,0 +1,13 @@
+ansicolors==1.1.8
+boto3==1.5.32
+botocore==1.8.50
+docutils==0.16
+jmespath==0.9.3
+python-dateutil==2.8.1
+PyYAML==4.2b4
+s3transfer==0.1.13
```
Comment (0xdabbad00) [inline requirements.txt:8]: I don't think I ever directly use this library. I think it is a requirement for botocore maybe and gets automatically installed.

File: requirements-dev.txt [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+jmespath==0.9.3
+python-dateutil==2.8.1
+PyYAML==4.2b4
+s3transfer==0.1.13
+six==1.15.0
+elasticsearch[es6]==6.1.1
+elasticsearch_dsl[es6]==6.1.0
+elasticsearch[es1]==1.9.0
+elasticsearch_dsl[es1]==0.0.11
+nose==1.3.7
```
Comment (0xdabbad00) [inline requirements-dev.txt:14]: I think all this file needs that requirements.txt doesn't have, is just `nose` and `coverage`, maybe a few others.  I'm guessing this file was autogenerated somehow.

File: requirements.txt (hunk 3 above)
Comment (kmcquade) [reply requirements.txt:8]: Yes, those are automatically installed. I generated this from running pip install for the exact versions mentioned in `setup.py`, then `pip freeze` and putting the exact ones needed in requirements.txt

File: requirements-dev.txt (hunk 4 above)
Comment (kmcquade) [reply requirements-dev.txt:14]: Oh. Good point. I kind of combined them together. 

I also included autopep8 and flake8, which were mentioned in the setup.py file.

File: .github/workflows/publish.yml (hunk 1 above)
Comment (kmcquade) [reply .github/workflows/publish.yml:30]: I suppose? 

I just tried running `python setup.py sdist bdist_wheel` without the `-m` flag and it wouldn't work, and then when I did it with the `-m` flag (as I've seen in other repositories) it worked as expected.

Comment (0xdabbad00) [reply .github/workflows/publish.yml:30]: With the `-m` flag I get:
```
python: Error while finding module specification for 'setup.py' (ModuleNotFoundError: __path__ attribute not found on 'setup' while trying to find 'setup.py')
```

This is on OSX with Python 3.7.6

File: requirements.txt (hunk 2 above)
Comment (kmcquade) [reply requirements.txt:10]: Cool I can remove that.

File: .github/workflows/publish.yml (hunk 1 above)
Comment (kmcquade) [reply .github/workflows/publish.yml:30]: Did you do that after `python3 -m venv ./venv && source venv/bin/activate`?

Comment (0xdabbad00) [reply .github/workflows/publish.yml:30]: Yes, from within the virtualenv.

Comment (jordan-wright) [reply .github/workflows/publish.yml:30]: FWIW, this one came from Github's own [starter workflows](https://github.com/actions/starter-workflows/blob/main/ci/python-publish.yml#L30). As such, I'd anticipate it working with Github Actions so I'd probably err away from modifying it.

Comment (kmcquade) [reply .github/workflows/publish.yml:30]: Will do. Apologies; I'll fix that shortly

File: requirements-dev.txt (hunk 4 above)
Comment (kmcquade) [reply requirements-dev.txt]: The new requirements-dev.txt file was generated via the following:

```bash
pip3 install nose coverage autoflake autopep8 pylint
pip3 freeze > requirements-dev.txt
```

The packages required by Cloudtracker itself are no longer included in the requirements-dev.txt file and remain in requirements.txt

File: requirements.txt (hunk 2 above)
Comment (kmcquade) [reply requirements.txt]: Removed in most recent commit

File: .github/workflows/publish.yml (hunk 1 above)
Comment (kmcquade) [reply .github/workflows/publish.yml]: Fixed in most recent commit

**Comment by 0xdabbad00** [comment]
Let's take a step back here for a second. What is the point of all this? What does "unit tests to GitHub actions" mean?  Does that mean when a PR comes in it automatically runs the unit tests?  I know for CloudMapper we have a webhook to Travis CI for that, but I guess nothing for something similar for this project.

I get that you want to do some additional development, so having a `requirements.txt` file makes sense, but I would just copy/paste these 4 lines to create that file: https://github.com/duo-labs/cloudtracker/blob/5f325c0300885813fcd136b6c33c075196330047/setup.py#L47

and for `requirements-dev.txt` I would just additionally add the libraries mentioned here https://github.com/duo-labs/cloudtracker/blob/5f325c0300885813fcd136b6c33c075196330047/setup.py#L42

**Comment by kmcquade** [comment]
> _What does "unit tests to GitHub actions" mean? Does that mean when a PR comes in it automatically runs the unit tests? I know for CloudMapper we have a webhook to Travis CI for that, but I guess nothing for something similar for this project._

Yep, you are correct. "Unit tests to GitHub actions" means that the unit tests themselves will actually execute in GitHub actions. In the current master branch, there are no automated tests, it just pushes new versions to PyPi when there is a new release made. 

* You can see the output of the actual tests here: https://github.com/duo-labs/cloudtracker/runs/971654655?check_suite_focus=true

* The automated testing to be performed via GitHub actions is called here:

https://github.com/duo-labs/cloudtracker/blob/2f66f5cc54e283236c14d9c9a9a9e31288c2c71c/.github/workflows/test.yml#L26

* It uses PyInvoke to run these tests, so you can do the testing using the same commands in both GitHub actions as well as locally. The PyInvoke for nosetests, for example, is here:

https://github.com/duo-labs/cloudtracker/blob/2f66f5cc54e283236c14d9c9a9a9e31288c2c71c/tasks.py#L104

* To run the tasks locally, you could do the following commands:

```bash
# Print out the available PyInvoke tasks from the tasks.py file
invoke -l
# Run nosetests
invoke unit.nose
# Lint 
invoke test.lint
# Run pytest with coverage
invoke test.pytest
```

> _I get that you want to do some additional development, so having a `requirements.txt` file makes sense, but I would just copy/paste... [lines directly from setup.py into requirements.txt and requirements-dev.txt]_

Oh, sure. I've always just run `pip freeze` to generate the requirements.txt so the dependencies and their dependencies are explicit, but I suppose if I put the direct dependencies only then they will just get installed anyway. That's fine.

**Comment by 0xdabbad00** [comment]
Awesome, this LGTM then. @jordan-wright @kylelady This is ready to merge.  I might clean up the `requirements.txt` file later to reduce it down to only the top level libraries, but otherwise this all looks like a solid improvement. Thank you @kmcquade

**Pull request merged**